server/apps/accounts/middlewares.py

```
@@ -0,0 +1,55 @@
+            raise AuthenticationFailed(
+                state.reason.name.lower() if state.reason else "unauthorized"
+            )
+
+        clerk_id = state.payload.get("sub")
+        if not clerk_id:
+            raise AuthenticationFailed("User identifier (sub) missing in token.")
+
+        try:
+            user = User.objects.get(username=clerk_id)
+        except ObjectDoesNotExist:
+            raise AuthenticationFailed("No such user")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)